.github/workflows/ci.yml

```diff
@@ -27,6 +27,8 @@ jobs:
       - run: cargo check --no-default-features
       - run: cargo check --no-default-features --features "use_alloc"
       - run: cargo test
+      - run: RUSTDOCFLAGS="-Dwarnings" cargo doc --all-features
+      - run: cargo clippy --all
```

Comment:
why not use a nightly target for clippy?

probably outside the scope of this PR, but why group CI targets by toolchain at all? Seems an odd way to slice it. Would be preferable to have this as separate, parallel jobs

Reply:
see #675 for an example of what i mean